File: src/main/scala/com/github/tetrisanalyzer/piecegenerator/PredictablePieceGenerator.scala
package com.github.tetrisanalyzer.piecegenerator

import com.github.tetrisanalyzer.piece.{PieceO, Piece}

/**
 * Generates a predictable sequence of pieces.
 */
class PredictablePieceGenerator(pieceSequence: List[Piece]) extends PieceGenerator {
  private val pieceSequenceIterator = pieceSequence.iterator

  def nextPieceNumber = {
    if (pieceSequenceIterator.hasNext)
      pieceSequenceIterator.next.number
    else
      new PieceO().number
  }
}

File: src/main/java/com/github/tetrisanalyzer/piecegenerator/PredictablePieceGenerator.java
package nu.tengstrand.tetrisanalyzer.piecegenerator;

import nu.tengstrand.tetrisanalyzer.piece.Piece;
import nu.tengstrand.tetrisanalyzer.piece.PieceO;

import java.util.Iterator;
import java.util.List;

public class PredictablePieceGenerator extends PieceGenerator {
    private Iterator<Piece> pieceSequenceIterator;

    public PredictablePieceGenerator(List<Piece> pieceSequence) {
        this.pieceSequenceIterator = pieceSequence.iterator();
    }

    @Override
    public int nextPieceNumber() {
        if (pieceSequenceIterator.hasNext()) {
            return pieceSequenceIterator.next().number();
        }
        return new PieceO().number();
    }
}
